mono/mini/mini-ops.h

```diff
@@ -185,6 +185,15 @@ MINI_OP(OP_FMOVE,	"fmove", FREG, FREG, NONE)
 MINI_OP(OP_VMOVE,   "vmove", VREG, VREG, NONE)
 MINI_OP(OP_RMOVE,	"rmove", FREG, FREG, NONE)
 
+/* Load the field at inst_offset from the vtype sreg1 */
+/* inst_imm is the field index of the field loaded */
+MINI_OP(OP_EXTRACTI4, "extracti4", IREG, VREG, NONE)
+MINI_OP(OP_EXTRACTI, "extracti", IREG, VREG, NONE)
+
+/* Set the field at inst_offset in the vtype sreg1 from sreg2, store the result into dreg */
+MINI_OP(OP_INSERTI4, "inserti4", VREG, VREG, IREG)
```

Comment:
I tried this design in the past and its interaction with CPROP is really tricky unless you force something like you did, which is disallow `dreg != sreg1`.

This PR lacks the code that generates them.

Reply:
The code will be added later. Probably only going to enable this for llvm, since it has a similar insert instruction so the conversion is trivial.